JS102/js_exercises102.js

```diff
@@ -0,0 +1,215 @@
+
+var printNumbers = function (x, y) {
+    var num = x
+    while (num <= y) {
+        console.log(num);
+        num ++;
+    }
+};
+
+printNumbers(1, 10);
+
+// Write two versions of the above function. 
+// One using a while loop and the other using a for loop.
+
+var printNumbers1 = function (x, y) {
+    for (var num = x; num <= y; num ++) {
+        console.log(num);
+    }
+};
+
+printNumbers1(1, 10);
+
+// Write a function printSquare which is given a size and 
+// prints a square of that size using asterisks.
+
+var printSquare = function (x) {
+    for (var num = 1; num <= x; num++) {
+        console.log('*'.repeat(x));
+    }
+};
+
+printSquare(5);
+
+// Write function printBox which is given a width and 
+// height and prints a hollow box of those given dimensions.
+
+var printBox = function (x, y) {
+    console.log('*'.repeat(x))
+    for (var height=1; height <= (y - 2); height++) {
+        console.log('*' + ' '.repeat(x-2) + '*');
+    }
+    console.log('*'.repeat(x))
+};
+
+printBox(6, 4);
+
+// Write a function printBanner which is given some text 
+// and prints a banner with a border surrounding the text. 
+// The border has to stretch to the length of the text.
+
+var printBanner = function (text) {
+    console.log('*'.repeat(text.length + 2));
+    console.log(`*${text}*`);
+    console.log('*'.repeat(text.length + 2));
+};
+
+printBanner('Welcome to DigitalCrafts');
+
+// Write a function factors which is given a number 
+// and returns an array containing all its factors.
+
+var factors = function (num) {
+    var factorList = [];
+    var x = 1;
+    while (x <= num) {
+        if (num % x == 0) {
+            factorList.push(x);
+        }
+        x++;
+    } 
+return factorList;
+};
+
+console.log(factors(6));
+
+// Write a function cipher which is given a string, 
+// an offset, and returns the Caesar cipher of the string.
+
+var plaintext = 'Genius without education';
+var ciphertext = '';
+var shift = 11;
+
+var cipher = function (plaintext, shift) {
+    for (var i = 0; i < plaintext.length; i++) {
+        var newChar = plaintext.charCodeAt(i);
+	    var shiftChar = newChar + shift;
+	    ciphertext += String.fromCharCode(shiftChar);
+	}
+return ciphertext;
+};
+
+console.log(cipher(plaintext, shift));
+
+// Write a function decipher which is given a string, an offset, 
+// and returns the original message.
+
+var shiftedText = ciphertext;
+var shift1 = -11;
+var fixed = '';
+
+var decipher = function (shiftedText, shift1) {
+    for (var i = 0; i < shiftedText.length; i++) {
+        var newChar = shiftedText.charCodeAt(i);
+        var shiftChar = newChar + shift1;
+        fixed += String.fromCharCode(shiftChar);
+    }
+return fixed;   
+};
+
+console.log(decipher(shiftedText, shift1));
+
+
+// Write a function leetspeak which is given a string, and 
+// returns the leetspeak equivalent of the string.
+
+
+var leetString = 'LEET';
+
+var leetSpeek = function (str) {
+    var str = str.toLowerCase();
```

Comment:
Does this code work? I did the same way and it only reads the last code of the function. I have figured the way to work. I will mention you in my branch so you can review them.